src/client.ts

```diff
@@ -155,36 +143,33 @@ export class Client {
     return this.#query(request.toQuery(headers));
   }
 
-  #getError(e: any): ServiceError | ProtocolError | NetworkError | ClientError {
-    // see: https://axios-http.com/docs/handling_errors
-    if (e.response) {
+  #getError(e: any): ClientError | NetworkError | ProtocolError | ServiceError {
+    // the error was already handled by the driver
+    if (
+      e instanceof ClientError ||
+      e instanceof NetworkError ||
+      e instanceof ProtocolError ||
+      e instanceof ServiceError
+    ) {
+      return e;
+    }
+
+    // the HTTP request succeeded, but there was an error
+    if (isHTTPResponse(e)) {
       // we got an error from the fauna service
-      if (isQueryFailure(e.response.data)) {
-        const failure = e.response.data;
-        const status = e.response.status;
+      if (isQueryFailure(e.body)) {
+        const failure = e.body;
+        const status = e.status;
         return this.#getServiceError(failure, status);
       }
+
       // we got a different error from the protocol layer
       return new ProtocolError({
```

Comment:
we seem to be losing the distinction between protocol and network errors.

This is a bit troubling as network errors are usually retriable and in P2 we'd like to build retries on these.

Can we use the guidance here: https://fetch.spec.whatwg.org/#concept-network-error to correctly determine a network error?

Reply:
This goes back to me thinking of network errors having to be handled in the HTTPClient, where they have context about what state the request is in. In that case, `#getError` is not handling any network errors.

The spec link says all of the reasons that a request should be a network error, but it doesn't include the implementation. The promise returned from `fetch` will already reject in those cases, so we don't have to check for status code == 0, etc. 

I have to figure out how to always pass along the appropriate context to `#getError` if we're punting everything to the client. Axios errors conveniently provided that context, but `fetch` does not.

Reply:
discussed offline. moving some of the distictions to a P2.